### PR TITLE
import OIDC resources exported from mac-fc-infra repo

### PR DESCRIPTION
**Pull request by leslie-corbalt**
https://jiraent.cms.gov/browse/CMCSMACD-1866

I created a terraform/dev/account directory with resources for the provider, role, and policy imported from AWS.
These resources are now managed by this repo.
Removed unnecessary permissions.

Tested:
Successfully exported from mac-fc-infra repo.
Successfully imported from AWS.
`terraform plan` shows No Changes.
updated lock file

### CODE REVIEW

File: terraform/dev/account/variables.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
+variable "subject_claim_filters" {
```
Comment (ben-harvey) [inline terraform/dev/account/variables.tf:1]: Do all of these need to be variables, or could you just put them inline in `github-oidc.tf`? I think these values won't change often and I don't think this module will be consumed elsewhere. If you want to keep them as variables I think that getting rid of the defaults and setting the values in a `terraform.tfvars` file is clearer

File: terraform/dev/account/variables.tf [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  default     = ["sts.amazonaws.com"] # the default audience for the GitHub OIDC provider, see https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#adding-the-identity-provider-to-aws
+}
+
+variable "thumbprint_list" {
+  description = " A list of thumbprints for the OIDC identity provider's server certificate"
+  type        = list(string)
+  default     = ["6938fd4d98bab03faadb97b34396831e3780aea1", "1c58a3a8518e8759bf075b76b750d4f2df264fcd"] # see https://github.blog/changelog/2023-06-27-github-actions-update-on-oidc-integration-with-aws/ and https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc_verify-thumbprint.html
+}
+
+variable "add_read_only_access" {
```
Comment (ben-harvey) [inline terraform/dev/account/variables.tf:19]: this variable is unused

File: terraform/dev/account/variables.tf (hunk 1 above)
Comment (leslie-corbalt) [reply terraform/dev/account/variables.tf]: I agree. I like being clear. I removed the variables. All the variables. I put the values in-line.
BTW, AWS says thumbprints are no longer used with github provider (but the attribute is still required by terraform).
I removed all variables. `variables.tf` is no longer necessary.

File: terraform/dev/account/variables.tf (hunk 2 above)
Comment (leslie-corbalt) [reply terraform/dev/account/variables.tf]: Yes. Removed. I removed `variables.tf`.

Comment (leslie-corbalt) [reply terraform/dev/account/variables.tf]: `terraform plan` shows No Changes.

**Review by ben-harvey — APPROVED** [review]
LGTM!

**Pull request merged**